src/AdoNet/Shared/Storage/DbExtensions.cs

```diff
@@ -7,12 +7,24 @@
 using System.Threading.Tasks;
 using System.Reflection;
 
-namespace Orleans.SqlUtils
+#if CLUSTERING_ADONET
+namespace Orleans.Clustering.AdoNet.Storage
+#elif PERSISTENCE_ADONET
+namespace Orleans.Persistence.AdoNet.Storage
+#elif REMINDERS_ADONET
+namespace Orleans.Reminders.AdoNet.Storage
+#elif STATISTICS_ADONET
+namespace Orleans.Statistics.AdoNet.Storage
+#elif TESTER_SQLUTILS
+namespace Orleans.Tests.SqlUtils
+#else
+// No default namespace intentionally to cause compile errors if something is not defined
+#endif
 {
     /// <summary>
     /// Contains some convenience methods to use in conjunction with <see cref="IRelationalStorage">IRelationalStorage</see> and <see cref="RelationalStorage">GenericRelationalStorage</see>.
     /// </summary>
-    public static class DbExtensions
+    internal static class DbExtensions
```

Comment:
Just a note about this and other changes to visibility, these can be used as a micro-ORM like Dapper, for instance. If someone has done that, this is a breaking change. If someone wish to do so, it's not possible anymore after this change.

Reply:
Yes we're aware of that, but this is the solution we chose to share files between pieces.